Google.PowerShell/BigQuery/BqJob.cs

```diff
@@ -415,12 +503,39 @@ public Apis.Bigquery.v2.Data.Job DoLoad()
 
         /// <summary>
         /// Extract Job main processing function.
+        /// *This is written using Apis.BigQuery.v2 becuase Cloud.BigQuery did not 
+        /// support Extract opertations at the time of writing.
         /// </summary>
         public Apis.Bigquery.v2.Data.Job DoExtract()
         {
             throw new NotImplementedException(
                 "Extract jobs are not implemented yet.  Use Get-BqTabledata instead.");
         }
+
+        /// <summary>
+        /// This function waits for a job to reach the "DONE" status and then returns.
+        /// </summary>
+        /// <param name="job">Job to poll for completion.</param>
+        public Apis.Bigquery.v2.Data.Job PollForCompletion(Apis.Bigquery.v2.Data.Job job)
```

Comment:
So in the process of transplanting all of the polling code and upgrading it, I got a weird error.  My code is currently:

**
// Check if the user is requesting a synchronous operation.
            if (PollUntilComplete)
            {
                //new code
                BigQueryJob bqj = Client.GetJob(result.JobReference);
                bqj.PollQueryUntilCompleted();
                result = bqj.Resource;
                //old code
                //result = PollForCompletion(result);
                //result = Client.GetJob(result.JobReference).Resource;
            }
**

When the old code is active, the tests all run fine.  however, when I swap to the new code (which is functionally the same), I get the error:
"Message[Job ~:job_~ is not a query job]"

I think this may have something to do with the fact that the veneer can't handle Copy, Load, and Extract jobs at the moment, so I'm going to leave the old code as what gets run, and the new code commented out.  This way, I can re-test when I have Load and Extract written to see if I can learn any more about this.  Also the code that I wrote works, so if I can't get the veneer polling to work, it's still functional.